public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,384 @@
+  const dispatch = useDispatch();
+  const detectorId: string = get(props, 'match.params.detectorId', '');
+
+  const adState = useSelector((state: AppState) => state.ad);
+  const allDetectors = adState.detectors;
+  const errorGettingDetectors = adState.errorMessage;
+  const detector = allDetectors[detectorId];
+
+  const [isStoppingDetector, setIsStoppingDetector] = useState<boolean>(false);
+  const [
+    historicalDetectorDetailModalState,
+    setHistoricalDetectorDetailModalState,
+  ] = useState<HistoricalDetectorDetailModalState>({
+    isOpen: false,
+    action: undefined,
+  });
+  const callout = getCallout(detector, isStoppingDetector);
+
+  useEffect(() => {
+    if (
+      errorGettingDetectors &&
+      errorGettingDetectors.includes(SINGLE_DETECTOR_NOT_FOUND_MSG)
+    ) {
+      props.history.push('/historical-detectors');
+    }
+  }, [errorGettingDetectors]);
+
+  useEffect(() => {
+    if (detector) {
+      core.chrome.setBreadcrumbs([
+        BREADCRUMBS.ANOMALY_DETECTOR,
+        BREADCRUMBS.HISTORICAL_DETECTORS,
+        { text: detector ? detector.name : '' },
+      ]);
+    }
+  }, [detector]);
+
+  const fetchDetector = async () => {
+    try {
+      await dispatch(getDetector(detectorId));
+    } catch {
+      core.notifications.toasts.addDanger(
+        'Unable to find the historical detector'
+      );
+    }
+  };
+
+  // Try to get the detector initially
+  useEffect(() => {
+    if (detectorId) {
+      fetchDetector();
+    }
+  }, []);
+
+  // If detector is initializing or running: keep fetching every 10 seconds to quickly update state/results/percentage bar, etc.
+  useEffect(() => {
+    if (
+      detector?.curState === DETECTOR_STATE.RUNNING ||
+      detector?.curState === DETECTOR_STATE.INIT
+    ) {
+      const intervalId = setInterval(
+        fetchDetector,
+        HISTORICAL_DETECTOR_RESULT_REFRESH_RATE
+      );
+      return () => {
+        clearInterval(intervalId);
+      };
+    }
+  }, [detector]);
+
+  const handleDeleteAction = () => {
+    setHistoricalDetectorDetailModalState({
+      ...historicalDetectorDetailModalState,
+      isOpen: true,
+      action: HISTORICAL_DETECTOR_ACTION.DELETE,
+    });
+  };
+
+  const handleEditAction = () => {
+    detector?.curState === DETECTOR_STATE.RUNNING ||
+    detector?.curState === DETECTOR_STATE.INIT
+      ? setHistoricalDetectorDetailModalState({
+          ...historicalDetectorDetailModalState,
+          isOpen: true,
+          action: HISTORICAL_DETECTOR_ACTION.EDIT,
+        })
+      : props.history.push(`/historical-detectors/${detectorId}/edit`);
+  };
+
+  const onStartDetector = async () => {
+    try {
+      await dispatch(startDetector(detectorId));
+      core.notifications.toasts.addSuccess(
+        `Successfully started the historical detector`
+      );
+    } catch (err) {
+      core.notifications.toasts.addDanger(
+        `There was a problem starting the historical detector ${detector?.name}: ${err}`
```

Comment:
may consider using `prettifyErrorMessage()` like [this](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/ec42b9fa6b62c7f7158b1178535bbcf5e519dbc6/public/pages/createDetector/containers/CreateDetector.tsx#L138-L140) to show prettier message in case of permission issue. Same comment for `err` case of StopDetector/DeleteDetector below

Reply:
Thanks, agreed this is better. Will change for the existing historical detector actions.

Reply:
Fixed in latest commit